crates/apollo-compiler/test_data/diagnostics/0092_recursive_fragment_spread.txt

```diff
@@ -1,12 +1,23 @@
 Error: `fragA` fragment cannot reference itself
-    ╭─[0092_recursive_fragment_spread.graphql:13:1]
+    ╭─[0092_recursive_fragment_spread.graphql:14:1]
     │
- 13 │ fragment fragA on Human { name, ...fragB }
+ 14 │ fragment fragA on Human { name, ...fragB }
     │ ───────┬──────  
     │        ╰──────── recursive fragment definition
     │ 
- 15 │ fragment fragC on Human { name, ...fragA }
+ 16 │ fragment fragC on Human { name, ...fragA }
     │                                 ────┬───  
     │                                     ╰───── refers to itself here
```

Comment:
The "refers to itself" message should probably be tweaked

Reply:
Should the diagnostic even contain a `Vec` of locations to show the complete cycle?

Reply:
This is easy to do currently. would be helpful to have more context lines :/
```
Error: `cycle` fragment cannot reference itself
    ╭─[q.graphql:9:1]
    │
  9 │ fragment cycle on __Type {
    │ ───────┬──────
    │        ╰──────── recursive fragment definition
    │
 11 │   ...frag
    │   ───┬───
    │      ╰───── `cycle` references `frag` here...
    │
 17 │     ...cycle
    │     ────┬───
    │         ╰───── `frag` circularly references `cycle` here
```

Reply:
It would be ideal here if, in TTY formatting, `cycle` in the final label uses the same colour as `fragment cycle` in the primary label 🙈